tools/cask/main.go

```diff
@@ -158,88 +169,83 @@ func Run(ctx context.Context, id string, dockerClient *client.Client) {
 	}
 }
 
-// GenerateBinds Create the bind mounts for input/output
-func GenerateBinds(outputPath string, policyPath string) []string {
-	// Policy
-	policy, err := filepath.Abs(policyPath)
-	if err != nil {
-		log.Fatalf("Unable to load policy. %v", err)
-	}
-
-	containerPolicy := containerHome + filepath.Base(policy)
-	binds := []string{
-		policy + ":" + containerPolicy + ":ro",
-	}
+// generateBinds Create the bind mounts for input/output
+func generateBinds(args []string) []string {
+	var binds []string
+
+	// Loop through all args looking for paths
+	// so we can create bind mounts and rewrite the arg
+	for i, arg := range args {
+		if isPath(arg) {
+			containerPath := containerHome + filepath.Base(arg)
+			absPath, err := filepath.Abs(arg)
+			if err == nil {
+				binds = append(binds, absPath+":"+containerHome+filepath.Base(absPath)+":rw")
+			}
 
-	// Output Path
-	if outputPath != "" {
-		outputPath, err = filepath.Abs(outputPath)
-		if err != nil {
-			log.Fatalf("Unable to parse output path. %v", err)
+			args[i] = containerPath
 		}
 
-		binds = append(binds, outputPath+":"+containerHome+"output:rw")
 	}
 
 	// Azure CLI support
-	azureCliConfig := GetAzureCliConfigPath()
+	azureCliConfig := getAzureCliConfigPath()
 	if azureCliConfig != "" {
 		// Bind as RW for token refreshes
 		binds = append(binds, azureCliConfig+":"+containerHome+".azure:rw")
 	}
 
 	// AWS config
-	awsConfig := GetAwsConfigPath()
+	awsConfig := getAwsConfigPath()
 	if awsConfig != "" {
 		binds = append(binds, awsConfig+":"+containerHome+".aws:ro")
 	}
 
-	return binds
-}
-
-// SubstitutePolicy Fix the policy arguments
-func SubstitutePolicy(args []string) string {
-	if len(args) == 0 ||
-		strings.EqualFold(args[0], "schema") ||
-		strings.EqualFold(args[0], "version") {
-		return ""
+	// Default cache location
+	if !funk.Any(funk.Intersect(args, []string{"-f", "--cache"})) {
+		cacheDefault := getFolderFromHome(".cache")
+		binds = append(binds, cacheDefault+":"+containerHome+".cache:rw")
 	}
 
-	originalPolicy := args[len(args)-1]
-	args[len(args)-1] = containerHome + filepath.Base(originalPolicy)
-
-	return originalPolicy
+	return binds
 }
 
-// SubstituteOutput Fix the output arguments
-func SubstituteOutput(args []string) string {
+func processOutputArgs(argsp *[]string) {
 	var outputPath string
+	args := *argsp
+
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "-s") || strings.HasPrefix(arg, "--output-dir") {
+			// normalize argument separator
+			if strings.HasPrefix(arg, "-s=") || strings.HasPrefix(arg, "--output-dir=") {
```

Comment:
I think the normalization of these arguments for things like `--metrics` and `--log-group` could leverage some of this logic.

Reply:
Yeah I didn't stress over any of those since I couldn't think of any values they might have which we need to know about beyond straight pass through  (e.g. no local paths or envs)